app/shop.py

```diff
@@ -0,0 +1,27 @@
+from datetime import datetime
+
+
+class Shop:
+    def __init__(self, name, location, products):
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def calculate_total_cost(self, product_cart):
+        total_cost = 0
+        for product, quantity in product_cart.items():
+            if product in self.products:
+                total_cost += self.products[product] * quantity
+            else:
+                return float('inf')

```

Comment:
Returning `float('inf')` when a product is not found in the shop's inventory might not be the best approach. Consider handling this case more gracefully, perhaps by notifying the user which product is unavailable.